command.go

```diff
@@ -1672,7 +1672,7 @@ func (c *Command) Parent() *Command {
 func (c *Command) mergePersistentFlags() {
 	c.updateParentsPflags()
 	c.Flags().AddFlagSet(c.PersistentFlags())
-	c.Flags().AddFlagSet(c.parentsPflags)
+	c.Flags().AddParentFlagSet(c.parentsPflags)
```

Comment:
Needs unit testing.

Also, not sure whether changing the default behavior is allowed. Probably this should be configurable.